File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Reflection;

namespace WindowsApplication1 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            list = new BindingList<MyObject>();
            
            for (int i = 0; i < 10; i++) {
                list.Add(new MyObject(i.ToString()));
            }
            
            gridControl1.DataSource = list;

        }


        BindingList<MyObject> list;

        private void simpleButton1_Click(object sender, EventArgs e) {
			using(XtraForm1 f = new XtraForm1(list)) {
				f.BindingContext = this.BindingContext;
				if(f.ShowDialog(this) == System.Windows.Forms.DialogResult.OK) {
					CurrencyManager cm = (CurrencyManager)this.BindingContext[list];
					cm.EndCurrentEdit();
				}
			}
        }

    }

    public class MyObject: INotifyPropertyChanged {
        public MyObject(string str) {
            field = str;
        }
        private string field;
        public string Field {
            get { return field; }
            set {
                field = value;
                NotifyPropertyChanged("Field");
            }
        }


        #region INotifyPropertyChanged Members

        private void NotifyPropertyChanged(String info) {
            if (PropertyChanged != null) {
                PropertyChanged(this, new PropertyChangedEventArgs(info));
            }
        }

        public event PropertyChangedEventHandler PropertyChanged;

        #endregion
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports System.Reflection

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			list = New BindingList(Of MyObject)()

			For i As Integer = 0 To 9
				list.Add(New MyObject(i.ToString()))
			Next i

			gridControl1.DataSource = list

		End Sub


		Private list As BindingList(Of MyObject)

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			Using f As New XtraForm1(list)
				f.BindingContext = Me.BindingContext
				If f.ShowDialog(Me) = System.Windows.Forms.DialogResult.OK Then
					Dim cm As CurrencyManager = CType(Me.BindingContext(list), CurrencyManager)
					cm.EndCurrentEdit()
				End If
			End Using
		End Sub

	End Class

	Public Class MyObject
		Implements INotifyPropertyChanged
		Public Sub New(ByVal str As String)
			field_Renamed = str
		End Sub
		Private field_Renamed As String
		Public Property Field() As String
			Get
				Return field_Renamed
			End Get
			Set(ByVal value As String)
				field_Renamed = value
				NotifyPropertyChanged("Field")
			End Set
		End Property


		#Region "INotifyPropertyChanged Members"

		Private Sub NotifyPropertyChanged(ByVal info As String)
			RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(info))
		End Sub

		Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

		#End Region
	End Class
End Namespace